### PR TITLE
Frontend/Kraken: Add continuous logging for better Extensions debugging

### PR DESCRIPTION
**Pull request by JoaoMario109**
* Add continuous logging in backend
* Add heartbeats in streamer to force data to frontend
* Add auto reconnect in frontend

### CODE REVIEW

File: core/frontend/src/views/ExtensionManagerView.vue [hunk 1]
```diff
@@ -410,61 +431,58 @@ export default Vue.extend({
...
-            if (!output) {
+            if (result) {
+              lastDecode = ansi.ansi_to_html(result)
+              this.log_info_output = `Logs for ${extension.name}`
+              this.$set(this, 'log_output', outputBuffer + lastDecode)
+            }
+            this.$nextTick(() => {
+              const logContainer = this.$refs.logContainer as HTMLElement
+              if (logContainer) {
+                logContainer.scrollTop = logContainer.scrollHeight
```
Comment (patrickelectric) [inline core/frontend/src/views/ExtensionManagerView.vue:465]: Can we avoid scrolling to the bottom every time that a new message is received when the scroll is not in the bottom ?
The use case is that if someone is trying to look at the logs, it should be able to look to previous log messages when new ones are being received.

**Review by patrickelectric — CHANGES_REQUESTED** [review]

File: core/services/kraken/api/v1/routers/index.py [hunk 2]
```diff
@@ -37,10 +37,19 @@ async def list_containers() -> Any:
...
-async def log_containers(container_name: str) -> StreamingResponse:
-    return StreamingResponse(
-        timeout_streamer(ContainerManager.get_container_log_by_name(container_name)), media_type="text/plain"
-    )
+async def log_containers(container_name: str, timeout: Optional[int] = None) -> StreamingResponse:
+    """
+    Fetch logs of a given container.
+    If timeout is provided, the stream will be closed after no log line is received for the given timeout.
+    If continuous is set to True, the stream will be kept open until the request be aborted by the client, even if the
+    container is stopped, it will keep trying to reconnect to the container log.
```
Comment (joaoantoniocardoso) [inline core/services/kraken/api/v1/routers/index.py:45]: There is no `continuous` parameter here.. is this docstring up to date??

Comment (JoaoMario109) [reply core/services/kraken/api/v1/routers/index.py]: Already removed, it was an old parameter, thanks!

File: core/frontend/src/views/ExtensionManagerView.vue (hunk 1 above)
Comment (JoaoMario109) [reply core/frontend/src/views/ExtensionManagerView.vue:471]: Added a Follow Logs checkbox to allow users to not auto scroll

**Review by patrickelectric — APPROVED** [review]

**Pull request merged**